csaf/src/metadata_retriever_new/mod.rs

```diff
@@ -0,0 +1,259 @@
+use crate::model::metadata::ProviderMetadata;
+use crate::source::HttpSourceError;
+use async_trait::async_trait;
+use sectxtlib::SecurityTxt;
+use url::{ParseError, Url};
+use walker_common::fetcher;
+use walker_common::fetcher::{Fetcher, Json};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Fetch error: {0}")]
+    Fetcher(#[from] fetcher::Error),
+    #[error("URL error: {0}")]
+    Url(#[from] ParseError),
+    #[error("securityTxtParseError error: {0}")]
+    SecurityTextError(#[from] sectxtlib::ParseError),
+    #[error("JSON parse error: {0}")]
+    Json(#[from] serde_json::Error),
+}
+
+impl From<Error> for HttpSourceError {
+    fn from(value: Error) -> Self {
+        match value {
+            Error::Fetcher(err) => Self::Fetcher(err),
+            Error::Url(err) => Self::Url(err),
+            Error::SecurityTextError(err) => Self::SecurityTextError(err),
+            Error::Json(err) => Self::Json(err),
+        }
+    }
+}
+
+#[derive(Clone)]
```

Comment:
It feels a bit like this overkill. Having a single retriever/detector is good enough I would say. The methods/functions looks good. So I think each of those structs could just share the same struct, with different methods.